cmd/util/ledger/migrations/migrator_runtime.go

```diff
@@ -56,6 +53,13 @@ func newMigratorRuntime(
 		runtime.NewCoverageReport(),
 	)
 
+	ri.GetOrLoadProgramFunc = func(
+		location runtime.Location,
+		load func() (*interpreter.Program, error),
+	) (*interpreter.Program, error) {
+		return load()
```

Comment:
Not sure if calling `load` here every time is safe, it probably needs to satisfy https://github.com/onflow/cadence/blob/699e3da675f6773559bf6ea0ee05963bca46b581/runtime/interface.go#L42-L54.

@janezpodhostnik Can we use the logic of FVM here?

Reply:
It looks like the default is already what we want: https://github.com/onflow/flow-go/blob/7969310cb95ecfabadeac24c25a836cb354ce7f2/cmd/util/ledger/util/migration_runtime_interface.go#L114, so maybe setting this function is just not needed? Or was adding it solving a particular problem?

Reply:
`m.Programs` is empty at this point

Reply:
Ah, right. So we need to initialize it in `MigrationRuntimeInterface` above:

I managed to come up with this so far, but can't see how to get a `storage.TransactionPreparer`:
```go
	ri := &util.MigrationRuntimeInterface{
		Accounts: accounts,
		Programs: environment.NewPrograms(
			tracing.NewTracerSpan(),
			util.NopMeter{},
			environment.NoopMetricsReporter{},
			nil, // TODO: how to get storage.TransactionPreparer?
			accounts,
		),
	}
```

Reply:
@onflow/flow-execution-and-verification does anyone know how we could get the proper `storage.TransactionPreparer` here?

Reply:
Thanks @janezpodhostnik! `NewDerivedBlockDataForScript` needs a `currentBlockId flow.Identifier`. What should be passed there?

Reply:
Tried fixing this in [59def00](https://github.com/onflow/flow-go/pull/5192/commits/59def00d5d58ed6398678a9a3f801d29b74d596a), please have a look 🙏

Reply:
That looks good. The block can be a empty identifier, its. not needed for this.